// zombie/lib/zombie/document.js
// Generated by CoffeeScript 1.4.0
var HTML, JSDOM, Scripts, createDocument;

JSDOM = require("jsdom");

Scripts = require("./scripts");

HTML = JSDOM.dom.level3.html;

createDocument = function(browser, window, referer) {
  var document, features, jsdomBrowser;
  features = {
    MutationEvents: "2.0",
    ProcessExternalResources: [],
    FetchExternalResources: [],
    QuerySelector: true
  };
  jsdomBrowser = JSDOM.browserAugmentation(HTML, {
    parser: browser.htmlParser
  });
  document = new jsdomBrowser.HTMLDocument({
    referrer: referer
  });
  if (browser.hasFeature("scripts", true)) {
    features.ProcessExternalResources.push("script");
    features.FetchExternalResources.push("script");
    Scripts.addInlineScriptSupport(document);
  }
  if (browser.hasFeature("css", false)) {
    features.FetchExternalResources.push("css");
  }
  if (browser.hasFeature("iframe", true)) {
    features.FetchExternalResources.push("iframe");
  }
  JSDOM.applyDocumentFeatures(document, features);
  Object.defineProperty(document, "window", {
    value: window
  });
  Object.defineProperty(document, "parentWindow", {
    value: window.parent
  });
  Object.defineProperty(document, "location", {
    get: function() {
      return window.location;
    },
    set: function(url) {
      return window.location = url;
    }
  });
  Object.defineProperty(document, "URL", {
    get: function() {
      return window.location.href;
    }
  });
  return document;
};

module.exports = createDocument;
